src/drive/ducks/files/Toolbar.jsx

```diff
@@ -136,11 +122,15 @@ class Toolbar extends Component {
         </Item>
         {notRootfolder && <hr />}
         {notRootfolder &&
-          (shared.withMe ? (
+          (isSharedWithMe ? (
             <Item>
               <a
                 className={classNames(styles['fil-action-delete'])}
-                onClick={() => leaveFolder(displayedFolder)}
+                onClick={() =>
+                  onLeave(displayedFolder).then(() =>
+                    trashFolder(displayedFolder)
```

Comment:
Small thing, but I think inline functions shouldn't be more than a one-liner, so maybe we could have a standalone function here? This would also allow to use `async/await`.

Reply:
I don't know if inline (one extra redirection) will make the code more readable in this case.
Still, you can open a discussion in https://github.com/cozy/cozy-guidelines/issues